File: lexilogicclient/src/main/java/org/amalgam/PlayerCallbackImpl.java
package org.amalgam;

import org.amalgam.ControllerException.InvalidRequestException;
import org.amalgam.UIControllers.PlayerCallbackPOA;

public class PlayerCallbackImpl extends PlayerCallbackPOA{
	private String username;
	private UpdateDispatcher updateDispatcher;

	@Override
	public String username() {
		return username;
	}

	@Override
	public void username(String newUsername) {
		this.username = newUsername;
	}

	@Override
	public void uiCall(String jsonString) throws InvalidRequestException {
		System.out.println(username + " " + jsonString);
		updateDispatcher.update(jsonString);
	}
	public void setControllerInterface(UpdateDispatcher updateDispatcher) {
		this.updateDispatcher = updateDispatcher;
	}
}

File: Client_Python/player_callback_impl.py
import org__POA
import org
from update_dispatcher import UpdateDispatcher


class PlayerCallbackImpl(org__POA.amalgam.UIControllers.PlayerCallback):
    username = ""
    update_dispatch = None

    def __init__(self): pass

    def get_username(self):
        return self.username

    def set_username(self, new_username):
        self.username = new_username

    def uiCall(self, json_string):
        self.update_dispatch.update(json_string)

    def controller_interface(self, update_dispatch):
        self.update_dispatch = update_dispatch


if __name__ == "__main__":
    player_callback_impl = PlayerCallbackImpl()
    print(dir(player_callback_impl))
    print(dir(org.amalgam.UIControllers.PlayerCallback))
    print(dir(org__POA.amalgam.UIControllers.PlayerCallback))
